pipeline_steps/common.groovy

```diff
@@ -433,4 +433,48 @@ def is_doc_update_pr(git_dir) {
   return is_doc_update_pr
 }
 
+/* Look for JIRA issue key in commit messages for commits in the source branch
+ * that aren't in the target branch.
+ * This function uses environment variables injected by github pull request
+ * builder and so can only be used for PR triggered jobs
+ */
+def get_jira_issue_key(repo_path="rpc-openstack"){
+  def key_regex = "^[a-zA-Z][a-zA-Z0-9_]+-[1-9][0-9]*"
```

Comment:
UG- is for unified gating issues, other projects will have their own identifiers. This code may get used for multiple projects so makes sense to be general.